### ISSUE TITLE
PLC2701 false positive for a private name import used in a type parameter list

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#24563]
### Summary

The documentation of [`import-private-name` (PLC2701)](https://docs.astral.sh/ruff/rules/import-private-name/) says it “ignores private name imports that are exclusively used in type annotations”, but that doesn’t include imports used in type parameter lists. Those are type annotations too, so it should ignore them too. [Example](https://play.ruff.rs/2d6e9b6d-c99e-4801-a9fb-5a9077cd601c):
```console
$ cat >plc2701.py <<'# EOF'
from __future__ import annotations
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from _typeshed import SupportsRichComparison
def f[T: SupportsRichComparison](x: T, y: T):
    return x < y or x > y
# EOF

$ ruff --isolated check --select PLC2701 --preview plc2701.py --output-format concise
plc2701.py:4:10: error[PLC2701] Private name import `_typeshed`
Found 1 error.
```

### Version

ruff 0.15.10 (252f76102 2026-04-09)
